controllers/front/SmailyCustomerCron.php

```diff
@@ -111,32 +88,117 @@ private function getUserData($customer)
         $syncronize_additonal = unserialize(Configuration::get('SMAILY_SYNCRONIZE_ADDITIONAL'));
         if (!empty($syncronize_additonal)) {
             foreach ($syncronize_additonal as $sync_data) {
-                $userdata[pSQL($sync_data)] = $customer[pSQL($sync_data)];
+                if (isset($customer[pSQL($sync_data)])) {
+                    $userdata[pSQL($sync_data)] = $customer[pSQL($sync_data)];
```

Comment:
Not sure, but doesn't escaping values sent to Smaily API using `pSQL` result in (possibly) malformed values?

For example escaping `User's field value` using `pSQL` would result in `User\'s field value`. This value will be JSON encoded to `User\\'s field value`. And in turn Smaily API decodes it as `User\'s field value`. Maybe it is better to omit the `pSQL` from Smaily API data collector and let Smaily worry about data clean up and field name normalization?

Reply:
Removed `pSQL` method and passing information as is. https://github.com/sendsmaily/smaily-prestashop-module/pull/10/commits/d96b432a1ae35ba93f4e0827d7c73453d56e113d